src/boot/efi/recovery.c

```diff
@@ -0,0 +1,23 @@
+/* SPDX-License-Identifier: LGPL-2.1-or-later */
+
+#include "recovery.h"
+#include "util.h"
+
+#define FALLBACK_BOOT_ENTRY L"ostree-1-lmp.conf\0"
+
+bool recovery_check_firmware_failure(void) {
+        EFI_SYSTEM_RESOURCE_TABLE *esrt_table = NULL;
+        EFI_SYSTEM_RESOURCE_ENTRY *esrt_entry = NULL;
+
+        esrt_table = find_configuration_table(MAKE_GUID_PTR(EFI_SYSTEM_RESOURCE_TABLE));
+        if (esrt_table) {
```

Comment:
we prefer early-exit over deep-nesting. i.e. change this line to `if (!esrt_table)\n return false;` so that the rest of the function doesn't have to be indented